src/components/profile-page/settings/profile-change-password.tsx

```diff
@@ -22,88 +24,39 @@ export const ProfileChangePassword: React.FC = () => {
   const [newPassword, setNewPassword] = useState('')
   const [newPasswordAgain, setNewPasswordAgain] = useState('')
 
-  const newPasswordValid = useMemo(() => {
-    return REGEX_VALID_PASSWORD.test(newPassword)
-  }, [newPassword])
-
-  const newPasswordAgainValid = useMemo(() => {
-    return newPassword === newPasswordAgain
-  }, [newPassword, newPasswordAgain])
-
-  const onChangeOldPassword = useCallback((event: ChangeEvent<HTMLInputElement>) => {
-    setOldPassword(event.target.value)
-  }, [])
-
-  const onChangeNewPassword = useCallback((event: ChangeEvent<HTMLInputElement>) => {
-    setNewPassword(event.target.value)
-  }, [])
-
-  const onChangeNewPasswordAgain = useCallback((event: ChangeEvent<HTMLInputElement>) => {
-    setNewPasswordAgain(event.target.value)
-  }, [])
+  const onChangeOldPassword = useOnInputChange(setOldPassword)
+  const onChangeNewPassword = useOnInputChange(setNewPassword)
+  const onChangeNewPasswordAgain = useOnInputChange(setNewPasswordAgain)
 
   const onSubmitPasswordChange = useCallback(
     (event: FormEvent) => {
       event.preventDefault()
-      changePassword(oldPassword, newPassword).catch(showErrorNotification('profile.changePassword.failed'))
+      doLocalPasswordChange(oldPassword, newPassword).catch(showErrorNotification('profile.changePassword.failed'))
     },
     [oldPassword, newPassword]
   )
 
+  const ready = useMemo(() => {
+    return (
+      oldPassword.trim() !== '' &&
```

Comment:
No. If we trim the value in the state, the user's input box doesn't match the stored state.
This `ready` only checks the conditions to disable or enable the submit button. It uses a memo to avoid unnecessary re-computations.

Reply:
So in theory a username / password with whitespace at the end is possible, but it just can't be all whitespace?

Reply:
exactly